streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -474,26 +468,24 @@ public void update(final Set<TopicPartition> topicPartitions, final Map<String,
 
     @Override
     public void closeAndRecycleState() {
-        suspend();
-        prepareCommit();
-        writeCheckpointIfNeed();
-
         switch (state()) {
-            case CREATED:
             case SUSPENDED:
                 stateMgr.recycle();
                 recordCollector.close();
 
                 break;
 
-            case RESTORING: // we should have transitioned to `SUSPENDED` already
-            case RUNNING: // we should have transitioned to `SUSPENDED` already
+            case CREATED:
+            case RESTORING:
+            case RUNNING:
             case CLOSED:
                 throw new IllegalStateException("Illegal state " + state() + " while recycling active task " + id);
             default:
                 throw new IllegalStateException("Unknown state " + state() + " while recycling active task " + id);
         }
 
+        // we cannot `clear()` the `PartitionGroup` in `suspend()` already, but only after committing,
+        // because otherwise we loose the partition-time information
         partitionGroup.clear();
```

Comment:
As we always suspend a task before closing (even for unclean closing), I think we can actually remove this call?